### ISSUE TITLE
New Linter Rule - set() must return a value

### ISSUE THREAD

**Issue by caseywatts** [ember-cli/eslint-plugin-ember#247]
having a linter rule for this would help prevent some bugs
(at least until the behavior changes to be what people expect)
https://github.com/emberjs/rfcs/pull/79

Bad:
```js
height: 100,
goldenRatioWidth: computed('height', {
  get() {
    return this.get('height') * 1.618;
  },
  set(_, value) {
    this.set('height', value / 1.618);
  }
})
```

Good
```js
height: 100,
goldenRatioWidth: computed('height', {
  get() {
    return this.get('height') * 1.618;
  },
  set(_, value) {
    const newHeight = value / 1.618;
    this.set('height', newHeight);
    return newHeight;
  }
})
```

`set()` must have a `return`.

(I mistakenly created this over [here](https://github.com/ember-cli/ember-cli-eslint/issues/232) first)

**Comment by Turbo87** [comment]
@caseywatts my naive idea would be this:

In the AST look for a `CallExpression` node with an `Identifier` of `computed. If the last `argument` is a `ObjectExpression` look for a `Property` with an `Identifier` of `set` and check if the `value` is a `FunctionExpression`. Now iterate over the `body` contents and check for any `ReturnStatement` nodes and if you can't find any you can report a lint failure.

You can use https://astexplorer.net/ to view the AST corresponding to any JS code you put in.

**Comment by clcuevas** [comment]
@caseywatts Pretty much what @Turbo87 summarized. Wish I knew of `astexplorer` in my previous PRs. It would have helped a lot!

Good luck!
